### ISSUE TITLE
Cannot access the si_pid field from waitid() call.

### ISSUE THREAD

**Issue by gatoWololo** [rust-lang/libc#1381]
Hello!

I'm trying call `waitid()` from [libc](https://docs.rs/libc/0.2.55/libc/fn.waitid.html) like:
```rust
let mut siginfo: libc::siginfo_t = unsafe { std::mem::zeroed() };
let ignored = 0;
libc::waitid(libc::P_ALL, ignored, &mut siginfo as *mut libc::siginfo_t , libc::WNOWAIT | libc::WEXITED | libc::WSTOPPED )
```

I use `waitid()` instead of `waitpid()` or `wait()` since I want the `WNOWAIT` flag. According to `man 2 waitid`:
```
Upon  successful  return,  waitid()  fills  in the following fields of the siginfo_t
       structure pointed to by infop:

       si_pid      The process ID of the child.
```

However, looking at the declaration of `sigino_t` for [Rust libc](https://docs.rs/libc/0.2.55/libc/struct.siginfo_t.html), it seems there is no `si_pid` field available. I know this is related to how sigino_t is defined #716.

Currently I'm linking out to C to get the correct field. It's ugly but it works.
```C
pid_t getPid(siginfo_t* infop) {
  return infop->si_pid;
}
```

I would like to transmute/cast the struct in Rust, but I have no idea what the `si_pid`-containing version of `siginfo_t` looks like.

**Comment by gnzlbg** [comment]
I think we should just add methods to `siginfo` in Rust that do the union access internally.

**Comment by joshtriplett** [comment]
I just ran into this myself. I need access to `si_pid` for use with `libc::WNOHANG`: when calling `waitid` with `WNOHANG`, you have to check if `si_pid == 0` to tell if you have a process status or if `waitid` has returned immediately because there aren't any waitable processes.

Long-term, I'd love to have the fields defined, but short-term, I'd settle for a method that returns the appropriate field value.

**Comment by joshtriplett** [comment]
https://github.com/rust-lang/libc/pull/1858 should fix this.

**Comment by JohnTitor** [comment]
I'm going to close this as fixed by #1858.
